packages/batch-execute/src/createBatchingExecutor.ts

```diff
@@ -16,6 +16,13 @@ export function createBatchingExecutor(
   const loadFn = createLoadFn(executor, extensionsReducer);
   const loader = new DataLoader(loadFn, dataLoaderOptions);
   return function batchingExecutor(request: ExecutionRequest) {
+    if (request.operationType == null) {
+      for (const def of request.document.definitions) {
+        if (isOperationDefinition(def)) {
+          request.operationType = def.operation;
```

Comment:
Should this function modify the request? Since it does, we do “as OperationTypeNode” scattered about, but maybe better to have a fallback from the document wherever so things have fewer type overrides?

Reply:
There’s a lot built around the presence of that operationType setting. Seemed simplest just to default it in absence once up front rather than repeatedly search for a default. However, if I understand your feedback there, you’re saying we should have a “operationTypeFromDocument” util that gets called here and in Delegate package rather than doing TypeScript “as” overrides…?

Reply:
Really good feedback. This has been cleaned up with a common `utils` method in 1642bdd9e67243d742bbb36e32207009eda61ee0. Feels a lot better!

Reply:
Actually we can easily get operation document with `getOperationAST` from graphql-js then get the operation type from it.